### PR TITLE
Feature/cbl 25

### PR DESCRIPTION
**Pull request by bmeike**
Make sure that all CouchbaseLiteExceptions have error messages.

### CODE REVIEW

**Comment by pasin** [comment]
FYI: checkStyle failed.

**Comment by bmeike** [comment]
Probably because the companion commit has not been checked in?
All tests pass on my machine.

**Comment by bmeike** [comment]
Rebased off the executor changes, for quick resolution.  Ready for review.

File: lib/src/shared/main/java/com/couchbase/lite/CouchbaseLiteException.java [hunk 1]
```diff
@@ -108,18 +133,18 @@ public CouchbaseLiteException(@NonNull String message, @NonNull Throwable cause,
      * @param cause the internal exception
      */
     public CouchbaseLiteException(@NonNull CBLInternalException cause) {
```
Comment (pasin) [inline lib/src/shared/main/java/com/couchbase/lite/CouchbaseLiteException.java]: I'm just curious how this works in Java when the method is public but the argument is not.

Comment (bmeike) [reply lib/src/shared/main/java/com/couchbase/lite/CouchbaseLiteException.java]: As you guessed, the method probably cannot be called.
It *could* be called, I suppose, if there were a public subtype of `CBLInternalException`
I think it is a mistake that *any* of these constructors are public.  Client code should *NOT* be creating new instances of `CouchbaseLiteException`.  The class and its accessors need to be public but I can't think of a single reason that the constructors should be.

**Pull request merged**